app/modules/key_management/secret_manager.py

```
@@ -22,18 +24,46 @@
+                status_code=500,
+                detail="SECRET_ENCRYPTION_KEY environment variable is not set",
+            )
+        # Ensure the key is in bytes. It must be a URL-safe base64-encoded 32-byte key.
+        try:
+            # This will raise if the key is invalid.
+            return Fernet(secret_key.encode("utf-8"))
+        except Exception as e:
+            raise HTTPException(
+                status_code=500,
+                detail=f"Invalid SECRET_ENCRYPTION_KEY: {str(e)}",
+            )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)